### PR TITLE
Relation utils

### PR DESCRIPTION
**Pull request by huwshimi**
Move all the relation utils to a separate module. This has been done so that these utils can replace the code duplicated between the canvas and the inspector and to ultimately work towards #1918.

### CODE REVIEW

File: jujugui/static/gui/src/app/utils/relation-utils.js [hunk 1]
```diff
@@ -0,0 +1,608 @@
...
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('relation-utils', function(Y) {
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/utils/relation-utils.js]: All the code in this module has been copied from elsewhere with no modification.

File: jujugui/static/gui/src/test/test_relation_utils.js [hunk 2]
```diff
@@ -0,0 +1,802 @@
...
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+
+describe('RelationUtils', function() {
```
Comment (huwshimi) [inline jujugui/static/gui/src/test/test_relation_utils.js]: All the tests in this file have been copied without modification.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5672/
Test PASSed.

**Comment by huwshimi** [comment]
QA:
- open the gui and try adding and removing relations on the canvas and through the inspector
- do any other relation interactions you can think of
- there should be no errors in the console

**Comment by hatched** [comment]
Thanks for splitting this out @huwshimi it would have been nice to de-YUI a lot of these as well but that can be done as a follow-up.

:+1: :ok:

**Comment by huwshimi** [comment]
Thanks @hatched :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**